### ISSUE TITLE
<Tab> insert un expected characters: "self"

### ISSUE THREAD

**Issue by roachsinai** [SirVer/ultisnips#1344]
**Expected behavior:**

Suppose current code of a python script is:

```
import os
os.
```

Then will show a completion list of member of python module os. Before I can use `<tab>` to choose which entry to use, and this is the expected behavior.

**Actual behavior:**

But now I press `<tab>` the content of this script change to

```
import os
osself.
```

and I found this issue was introduced by commit https://github.com/SirVer/ultisnips/commit/b974a13328071de45a85c62ab65c8bfed0142728 .

A gif to show this issue:

![Peek 2021-04-13 23-27](https://user-images.githubusercontent.com/9500049/114830037-5b109700-9dfe-11eb-92f8-de31fb14b3d8.gif)

**Steps to reproduce**

minimal_vimrc:

```
call plug#begin('~/.vim/plugged')

Plug 'ycm-core/YouCompleteMe'
Plug 'SirVer/ultisnips'
Plug 'honza/vim-snippets'

let g:UltiSnipsSnippetsDir="~/.vim/snip"
let g:UltiSnipsSnippetDirectories=["snip", "UltiSnips"]

" https://github.com/ycm-core/YouCompleteMe/issues/36#issuecomment-171966710
function! UltiSnips_Complete()
  call UltiSnips#ExpandSnippet()
  if g:ulti_expand_res == 0
    if pumvisible()
        return "\<C-n>"
    else
      call UltiSnips#JumpForwards()
      if g:ulti_jump_forwards_res == 0
        return "\<TAB>"
      endif
    endif
  endif
  return ""
endfunction

function! UltiSnips_Reverse()
  call UltiSnips#JumpBackwards()
  if g:ulti_jump_backwards_res == 0
    return "\<C-P>"
  endif

  return ""
endfunction

let g:UltiSnipsJumpForwardTrigger = "<tab>"
let g:UltiSnipsJumpBackwardTrigger="<s-tab>"

au InsertEnter * exec "inoremap <silent> " . g:UltiSnipsExpandTrigger . " <C-R>=UltiSnips_Complete()<cr>"
au InsertEnter * exec "inoremap <silent> " . g:UltiSnipsJumpBackwardTrigger . " <C-R>=UltiSnips_Reverse()<cr>"

call plug#end()
```

- **Operating System**: Manjaro Linux
- **Vim Version**: gvim 8.2.2653
- **UltiSnips Version**: lasted
- **Python inside Vim**: 3.9.2

**Releated issue**

#1325

**Comment by SirVer** [comment]
Thanks! Fixed in #1345.

**Comment by roachsinai** [comment]
Thank you so much for the quick patch!
